cl/citations/match_citations.py

```diff
@@ -278,13 +278,19 @@ def resolve_supra_citation(
     )
 
 
+@no_type_check
 def do_resolve_citations(
-    citations: List[CitationBase], citing_opinion: Opinion
+    citations: List[CitationBase], citing_document: Opinion | RECAPDocument
 ) -> Dict[MatchedResourceType, List[SupportedCitationType]]:
     # Set the citing opinion on FullCaseCitation objects for later matching
     for c in citations:
         if type(c) is FullCaseCitation:
-            c.citing_opinion = citing_opinion
+            if isinstance(citing_document, Opinion):
+                c.citing_opinion = citing_document
+            elif isinstance(citing_document, RECAPDocument):
+                c.citing_document = citing_document  # if the writing doing the citing is a RECAPDocument, then refer to it as a citing document.
```

Comment:
This is a bit confusing, and I should probably rename the parameter, but I am not sure what else to call it. "Citing thing"? lol

Reply:
`citing_object`?